protocol-units/sequencing/memseq/sequencer/src/lib.rs

```diff
@@ -444,6 +457,13 @@ pub mod test {
 			Err(anyhow::anyhow!("Mock pop_mempool_transaction"))
 		}
 
+		async fn gc_mempool_transactions(
+			&self,
+			_timestamp_threshold: u64,
+		) -> Result<(), anyhow::Error> {
+			Err(anyhow::anyhow!("Mock gc_mempool_transaction"))
```

Comment:
@mzabaluev Can you outline the approach you intend to take here again? That's the most important thing for me to verify currently.

Reply:
The GC routine makes a pass of the `mempool_transactions` CF in the key order, stopping at a key prefix that's obtained from the timestamp parameter. All transactions with preceding keys are removed. This works because the transaction timestamp is the leftmost part of the key. In the sequencer, the timestamp cutoff is chosen as twice the block forming period before the current time. The heuristic can be tweaked, of course.